### PR TITLE
More Python-SDK love

### PR DESCRIPTION
**Pull request by davidiw**
Inspired by our hackathon, I added support for wrapping the CLI to the Python SDK. Simply place the aptos cli in your path (`aptos`) or pass in an explicit path (`APTOS_CLI_PATH`).

This can then be used to:
* Start a local testnet
* Unit test move
* Compile move packages

Which in turn I used to make most of the Python examples easily testable -- next step is CI/CD somehow.

Also improved the chunking ability for the large package publisher, which makes it as good as it can be.

Once this lands, we should actually cut 0.7

### CODE REVIEW

**Comment by gregnazario** [comment]
@0xjinn may need to make more stringent input output tests around these commands

**Review by 0xmaayan — COMMENTED** [review]
a bit confused, and probably b/c I am not an expert in python, but how/who use this feature? Is it for the sdk developers/maintainers? or for external users that depends on the python sdk in their proejcts?

File: ecosystem/python/sdk/aptos_sdk/aptos_cli_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+from .account_address import AccountAddress
+from .async_client import FaucetClient, RestClient
+
+# Assume that the binary is in the global path if one is not provided.
+DEFAULT_BINARY = os.getenv("APTOS_CLI_PATH", "aptos")
+LOCAL_FAUCET = "http://127.0.0.1:8081"
+LOCAL_NODE = "http://127.0.0.1:8080/v1"
+
+# Assume that the node failed to start if it has been more than this time since the process started
+MAXIMUM_WAIT_TIME_SEC = 30
```
[the remark this answers is not in the document]
Comment (davidiw) [reply ecosystem/python/sdk/aptos_sdk/aptos_cli_wrapper.py:24]: should be done in like 1-5 seconds -- at 30 something bad probably happened...

**Comment by davidiw** [comment]
> a bit confused, and probably b/c I am not an expert in python, but how/who use this feature? Is it for the sdk developers/maintainers? or for external users that depends on the python sdk in their proejcts?

As an SDK developer, this makes it a LOT easier to do end-to-end testing of all the examples.

As an app developer, there was no easy way to actually facilitate end-to-end testing of your code without a complex setup.

So in effect it helps anyone that wants to get good integration testing as part of their development flow or ci-cd.

The closest thing we have to this would be the move e2e tests, which are great for us framework developers but other move developers are likely suffering. It could be conceived as  a precursor to hard hat. We likely want something like this in the Typescript SDK if it isn't already there.

**Comment by 0xmaayan** [comment]
> > a bit confused, and probably b/c I am not an expert in python, but how/who use this feature? Is it for the sdk developers/maintainers? or for external users that depends on the python sdk in their proejcts?
> 
> As an SDK developer, this makes it a LOT easier to do end-to-end testing of all the examples.
> 
> As an app developer, there was no easy way to actually facilitate end-to-end testing of your code without a complex setup.
> 
> So in effect it helps anyone that wants to get good integration testing as part of their development flow or ci-cd.
> 
> The closest thing we have to this would be the move e2e tests, which are great for us framework developers but other move developers are likely suffering. It could be conceived as a precursor to hard hat. We likely want something like this in the Typescript SDK if it isn't already there.

gotcha. yes we talked about having the option to compile a move module in the sdk very beneficial.

**Comment by xbtmatt** [comment]
Interesting, I hadn't even seen this yet before we talked today @0xmaayan, haha

To tack onto this a bit, I already thought the improvement on the build/compilation process was nice. But now that I think about it more, a large portion of dapp & wallet issues essentially come down to a lack of support for integration testing, since they're very frequently just serialization and tx builder issues.

This would make debugging these things (and ultimately standardizing wallet functionality) a lot easier, we should talk about it on thursday 😄

File: aptos-move/move-examples/large_packages/sources/large_packages.move [hunk 2]
```diff
@@ -4,11 +4,14 @@
 /// Currently this module does not support modules that are larger than 63KB as that is the maximum
 /// that can fit within a transaction and this framework does not split up individual modules.
```
Comment (xbtmatt) [inline aptos-move/move-examples/large_packages/sources/large_packages.move:5]: ah now I understand why I was confused today about the max size...you were referring to it supporting packages of any size, but the module size is still limited. although even that is only bc this doesn't handle splitting them up for you, right?

Comment (davidiw) [reply aptos-move/move-examples/large_packages/sources/large_packages.move:5]: well it was... but I didn't update the comments...

I generalized the metadata chunking logic and then applied it to modules, so now it supports any size.

Comment (xbtmatt) [reply aptos-move/move-examples/large_packages/sources/large_packages.move:5]: ahh right, you'd just have to call it multiple times

File: ecosystem/python/sdk/aptos_sdk/package_publisher.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 # The location of the large package publisher
 MODULE_ADDRESS: AccountAddress = AccountAddress.from_str(
-    "0xd20f305e3090a24c00524604dc2a42925a75c67aa6020d33033d516cf0878c4a"
+    "0xfa3911d7715238b2e3bd5b26b6a35e11ffa16cff318bc11471e84eccee8bd291"
```
Comment (gregnazario) [inline ecosystem/python/sdk/aptos_sdk/package_publisher.py:17]: Is this on mainnet?  testnet?  devnet?

**Review by gregnazario — APPROVED** [review]
Please write instructions for devnet / testnet / mainnet, and add an error message

File: ecosystem/python/sdk/examples/your_coin.py [hunk 4]
```diff
@@ -78,11 +79,7 @@ async def get_balance(
         return balance["data"]["coin"]["value"]
 
 
-async def main():
-    assert (
-        len(sys.argv) == 2
-    ), "Expecting an argument that points to the moon_coin directory."
-
+async def main(moon_coin_path: str):
```
[the remark this answers is not in the document]
Comment (davidiw) [reply ecosystem/python/sdk/examples/your_coin.py:82]: so the package is actually moon coin :grimacing:

File: ecosystem/python/sdk/CHANGELOG.md [hunk 5]
```diff
@@ -3,14 +3,16 @@
...
 ## 0.7.0
-- **[Breaking Change]**: Removed the `hex` function from `AccountAddress`. Instead of `addr.hex()` use `str(addr)`.
-- **[Breaking Change]**: The string representation of `AccountAddress` now conforms to [AIP-40](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-40.md).
-- **[Breaking Change]**: `AccountAddress.from_hex` and `PrivateKey.from_hex` have been renamed to `from_str`.
-- **[Breaking Change]**: Delete sync client
 - Port remaining sync examples to async (hello-blockchain, multisig, your-coin)
 - Updated token client to use events to acquire minted tokens
 - Update many dependencies and set Python 3.8.1 as the minimum requirement
 - Add support for an experimental chunked uploader
+- Add experimental support for the Aptos CLI enabling local end-to-end testing, package building, and package integration tests
```
[the remark this answers is not in the document]
Comment (davidiw) [reply ecosystem/python/sdk/CHANGELOG.md:10]: nope

File: aptos-move/move-examples/large_packages/large_package_example/sources/four.move [hunk 6]
```diff
@@ -33,10 +43,207 @@ module large_package_example::four {
...
+        let vec = vector::empty();
+        vector::append(&mut vec, inline_large_vector());
+        vector::append(&mut vec, inline_large_vector());
+        vector::append(&mut vec, inline_large_vector());
+        vector::append(&mut vec, inline_large_vector());
+        vector::append(&mut vec, inline_large_vector());
```
[the remark this answers is not in the document]
Comment (davidiw) [reply aptos-move/move-examples/large_packages/large_package_example/sources/four.move:51]: was trying to increase the size, might be worth a try again some point in the future lol

**Review by xbtmatt — APPROVED** [review]

**Pull request merged**